src/device_hub.cpp

```diff
@@ -17,22 +17,26 @@ namespace librealsense
         for (auto dev : devices)
         {
             auto data = dev->get_device_data();
-            for (auto uvc : data.uvc_devices)
+            for (const auto& uvc : data.uvc_devices)
             {
                 if (uvc.vid == vid || vid == 0)
                 {
                     result.push_back(dev);
                     break;
                 }
             }
+
+            if (data.tm2_devices.size() && (0== vid))
```

Comment:
Why vid==0?

Reply:
vid-based filtering is currently not supported for T265. Backend changes in tm2_info are required.
I made a semantic change to improve maintainability, also marked a TODO